src/main/java/br/com/moip/creditcard/MasterCreditCard.java

```diff
@@ -1,9 +1,29 @@
 package br.com.moip.creditcard;
 
+import java.util.Arrays;
+import java.util.List;
+
 public class MasterCreditCard {
 
+    public static List<String> MASTERCARD_NEW_RANGE = Arrays.asList("222100", "272099");
+
     public static boolean isBrandMaster(final String number) {
-        return number != null && number.matches("5[0-9]{15}");
+        return number != null && (number.matches("5[0-9]{15}") || isValidMasterBin(number));
+    }
+
+    private static boolean isValidMasterBin(String number) {
+        if (number.length() < 5) {
+            return false;
+        }
+
+        for (int i = 1; i < MASTERCARD_NEW_RANGE.size(); i += 2) {
+            int startingRange = Integer.valueOf(MASTERCARD_NEW_RANGE.get(0));
+            int endingRange = Integer.valueOf(MASTERCARD_NEW_RANGE.get(i));
```

Comment:
for (int i = 0...) {
get(i) e get(i+1), assim ele vai pegar as posições (0 e 1), (2 e 3), (4 e 5)...
Isso fará com que seu código ainda funcione se você quiser usar mais ranges no futuro.